contracts/SmartLoan.sol

```diff
@@ -122,19 +122,20 @@ contract SmartLoan is OwnableUpgradeable, PriceAwareUpgradeable, ReentrancyGuard
   function payBonus(uint256 _bonus) internal {
     if (_bonus < address(this).balance) {
       payable(msg.sender).transfer(_bonus);
-    } else {
+    } else if (address(this).balance != 0) {
       payable(msg.sender).transfer(address(this).balance);
     }
   }
 
 
   function selloutLoan() external onlyOwner {
-    uint256 debt = getDebt();
-    // Extra 10% to sellout all assets despite differences between an oracle's and exchange valuations.
-    uint256 targetPostSelloutAvaxValue = getTotalValue() * 11 / 10;
-    sellout(targetPostSelloutAvaxValue);
+    bytes32[] memory assets = supportedAssets.getAllAssets();
+    for (uint i = 0; i < assets.length; i++) {
+      nonSolventAssetSale(assets[i], getERC20TokenInstance(assets[i]).balanceOf(address(this)), 0);
+    }
 
-    if (address(this).balance < debt) {
+    uint256 debt = getDebt();
+    if (address(this).balance <= debt) {
       repay(address(this).balance);
```

Comment:
Imho, if after the full sellout balance is lower than debt we should raise an exception. 
We should only allow the sellout if a user could fully repay the debt. 
Otherwise, we will incentive manipulation of the DEX to avoid escaping a loan without repaying all the debt to pool.

Reply:
Good point, I changed this in the newest version